plugins/utils/navbuddy.nix

```diff
@@ -0,0 +1,175 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.navic;
+  helpers = import ../helpers.nix {inherit lib;};
+  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
+in {
+  options.plugins.navic =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-navbuddy";
+
+      package = helpers.mkPackageOption "nvim-navbuddy" pkgs.vimPlugins.nvim-navbuddy;
+
+      window = {
+        border = helpers.defaultNullOpts.mkBorder "rounded" "double" "solid" "none" ''
+          "rounded", "double", "solid", "none"  or an array with eight chars building up the border in a clockwise fashion
+          starting with the top-left corner. eg: { "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" }.
+        '';
+
+        # TODO: how to do attribute set here or string ?
+        size = helpers.defaultNullOpts.mkStr "60%" ''
+          Or table format example: { height = "40%", width = "100%"}
+        '';
+
+        # TODO: see above
+        position = helpers.defaultNullOpts.mkStr "50%" ''
```

Comment:
how to do attribute set here or string ?

Reply:
see above